docs/TECHNICAL_DESIGN.md

```diff
@@ -0,0 +1,155 @@
+## Technical Design
+
+Carrot follows the Facebook Flux pattern:
+
+<img src="./flux-diagram-white-background.png" style="width: 100%;" />
+
+_Image source: the [Flux repo](https://github.com/facebook/flux)_
+
+It's important to note, Flux isn't a particular software library, instead it's an architectural pattern that's been implemented now in countless React apps and in many React web frameworks.
+
+The key idea in Flux is having a unidirectional data flow:
+
+<img src="./flux-simple-f8-diagram-with-client-action-1300w.png" style="width: 100%;" />
+
+_Image source: [Flux concepts](https://github.com/facebook/flux/tree/master/examples/flux-concepts)_
+
+Here is some background information on Flux:
+
+- https://github.com/facebook/flux - The Flux repo (documentation of the pattern, as well as a JavaScript Dispatcher implemntation)
+- https://github.com/facebook/flux/tree/master/examples/flux-concepts - High-level Flux concepts
+- https://facebook.github.io/flux/docs/in-depth-overview.html - Flux in depth
+- https://github.com/kgann/cljs-flux - An example implementation of the Flux pattern in ClojureScript
+
+### Action Creators
+
+An action creator is called by a user interaction or an API request finishing.
+
+These action creators are located in `/src/oc/web/actions/`. They are losely
+organized based the data they will act on. So `/src/oc/web/actions/user.cljs`
+contains actions for a user to sign up, login, and change user profile
+information.
+
+
+### Simplified Data Model
+
+- Org
+  - has many Teams
+  - has many Sections
+
+- Team
+  - has many Users
+  - Authentication Settings
+  - Slack bot
+
+- User
+  - Slack info
+
+- Section
+  - Slack info 
+  - has many Posts
+  - has many authors
+  - has many viewers
+
+- Post
+  - has many Comments
+  - has many Reactions
+  - has many authors
+
+- Reaction
+  - has an author
+
+- Comment
+  - has many Reactions
+  - has an author
+  
+### Stores (changing the app state)
+
+Our action creators will dispatch data and an event using our dispatcher.
+We subscribe stores to particular events by creating a function with an arity
+matching the event and payload. Stores with interest in the same event/payload
+will have a separate register to the dispatcher.
+
+When parts of the app state are changed a derivative is changed and used to
+render the UI. Also UI components can read from a store to retrieve more information.
+
```

Comment:
@thepug Similarly I think it would be good to show code of a simple store or an example. In particular, should be clear: registering with the dispatcher, receiving an action, and changing a derivative.